torch/_subclasses/fake_tensor.py

```diff
@@ -547,6 +547,13 @@ class FakeTensor(torch.Tensor):
     fake_mode: "FakeTensorMode"
     constant: Optional[torch.Tensor]
 
+    @property
+    def device(self):
+        if self.fake_mode.in_kernel_invocation:
+            return torch.device("meta")
```

Comment:
it's probably cheap to construct meta device, but you could cache the construction

Reply:
This doesn't seem to matter at all.

With 

```
diff --git a/torch/_subclasses/fake_tensor.py b/torch/_subclasses/fake_tensor.py
index 0226e1526d5..891ea93d9d2 100644
--- a/torch/_subclasses/fake_tensor.py
+++ b/torch/_subclasses/fake_tensor.py
@@ -534,6 +534,9 @@ class FakeTensorConfig:
     debug = os.environ.get("TORCH_FAKE_TENSOR_DEBUG", False)
 
 
+META_DEVICE = torch.device("meta")
+
+
 class FakeTensor(torch.Tensor):
     """
     Meta tensors give you the ability to run PyTorch code without having to
@@ -550,7 +553,7 @@ class FakeTensor(torch.Tensor):
     @property
     def device(self):
         if self.fake_mode.in_kernel_invocation:
-            return torch.device("meta")
+            return META_DEVICE
         else:
             return self.fake_device
 

```

I get

```
(/home/ezyang/local/a/pytorch-env) [ezyang@devgpu020.ftw1 ~/local/a/pytorch (f13028ce)]$ python benchmarks/dynamo/timm_models.py --accuracy --timing --backend aot_eager --dynamic-shapes --float32 --only hrnet_w18
cuda eval  hrnet_w18                           PASS
TIMING: entire_frame_compile:55.28344 backend_compile:34.49498
STATS: call_* op count: 1369 | FakeTensor.__torch_dispatch__:4995 | FakeTensorMode.__torch_dispatch__:89985 | ProxyTorchDispatchMode.__torch_dispatch__:3010
```